### PR TITLE
Support multiple input patterns

### LINKED ISSUE #49: Getting match for single machine only
It's most likely an issue in upstream `gaze`, but just in case someone else stumbles upon it

I have my machines structured like

```
machinery
  src
    firstMachine
      firstMachine.ts
      index.ts
    secondMachine
      secondMachine.ts
      index.ts
```

Upon running `xstate-codegen machinery/src/**/**Machine.ts` I get match for the `firstMachine`, nothing else. And that machine isn't even using `@xstate/compiled` yet (maybe it should be skipped? 🤷‍♂️).

I actually tried logging `filesAsRecord` and it indeed gives a single result for some reason.

I am running this in WSL2, so it's not a Windows issue. But if you see some problem with my pattern, let me know. Otherwise, I will probably open the issue with `gaze`.

### PR DESCRIPTION
**Pull request by FredyC**
I kinda needed this for my project where machines are in different folders, so I went for it.

Sadly the `gaze` doesn't support multiple inputs and it would be annoying with callbacks. So I replaced it with `chokidar` which seems solid. I stumbled upon the issue where `ready` event is [triggered too soon](https://github.com/paulmillr/chokidar/issues/1011) so I had to drop some of the previous logging and checks, but nothing too serious imo.

Fixes #49

### CODE REVIEW

**Comment by FredyC** [comment]
Uh, no idea why changeset is not detected, I've added it there. Maybe force push is a problem? 🤷 Will see with a further changes based on CR.

**Comment by mattpocock** [comment]
@FredyC I've been thinking about dropping gaze for chokidar as it seems like a better choice in general. I'll review this and merge it with priority. Thanks so much for this.

File: packages/xstate-compiled/src/index.ts [hunk 1]
```diff
@@ -1,133 +1,105 @@
...
-  ) as string[];
-};
-
-if (!pattern) {
-  console.log('You must pass a glob, for instance "**/src/**.machine.ts"');
+if (patterns.length === 0) {
+  console.log(
+    'You must pass at least one glob, for instance "**/src/**.machine.ts"',
+  );
   process.exit(1);
```
Comment (mattpocock) [inline packages/xstate-compiled/src/index.ts:18]: Should we still exit 0?

Comment (FredyC) [reply packages/xstate-compiled/src/index.ts:18]: I think this is fine to be a failure, without any patterns it doesn't make sense to run this. I think you have confused this with the situation when no files are found as a match to patterns. Unfortunately, that's not easy to do with `chokidar`, I would have to unwrap all globs manually only for that check.

**Comment by mattpocock** [comment]
Tested locally, working wonderfully. Also fixes an annoying bug where it wasn't picking up new files added.

**Review by mattpocock — APPROVED** [review]

**Pull request merged**